extendr-api/src/robj/mod.rs

```diff
@@ -734,65 +733,143 @@ where
 {
     fn as_typed_slice(&self) -> Option<&'a [T]>
     where
-        Self: 'a,
-    {
-        None
-    }
+        Self: 'a;
 
     fn as_typed_slice_mut(&mut self) -> Option<&'a mut [T]>
     where
-        Self: 'a,
-    {
-        None
+        Self: 'a;
+}
+
+/// Trait for providing a rust typed pointer to an r type.
+/// R data are hidden behind [`SEXP`] opaque pointers with an identifier
+/// for the underlying r storage type. Thus this trait is only meaningful for
+/// [`SEXP`]-wrappers.
+trait SexpAs {
```

Comment:
Could this be `SexpAsPtr` for clarity?

Reply:
`SEXP` is already a pointer. So yes.. maybe that would work better :)